### PR TITLE
Add new permissions.

### LINKED ISSUE #55: New user roles
A number of new roles have been devised to further granulate authorization, and prevent the admin from having to do everything:

Projects
- A project member can create times for that project
- A project spectator can view all times associated with their project(s)
- A project manager can update and delete projects

Sitewide
- A spectator can view all times
- A manager can create projects; promote users to project manager; create, update, and delete activities; and add users.
- An admin can perform _all_ activities across the site.

### PR DESCRIPTION
**Pull request by MorganEPatch**
Fixes #55. I also propose changing "roles" to "permissions" in our documentation.

### CODE REVIEW

**Comment by athai** [comment]
Looks good. +1

File: source/draft_api.rst [hunk 1]
```diff
@@ -991,29 +997,25 @@ When attempting to delete a project or activity, it must not be referenced by a
...
 
-* a member has permission to write time entries
-* a spectator may view time entries
-* a project manager may update the project information
+* A project member has permission to write time entries for that project
+* A project spectator may view time entries for that project (e.g. ``GET /times?project=foo``)
+* A project manager may update the project information and delete a project, as well as add and remove members
+* A sitewide spectator may view all times
+* A sitewide manager may create projects, promote users to project manager, create, update, and delete activities, and create and delete users
+* An admin may perform any action across the site (including promoting sitewide managers)
```
Comment (pop) [inline source/draft_api.rst:1024]: Let's add a table that makes all of these permissions a bit easier to unpack.

File: source/draft_api.rst [hunk 2]
```diff
@@ -725,7 +729,7 @@ Response body:
     {
        "uri": "https://code.osuosl.org/projects/timesync",
        "name": "TimeSync API",
-       "slugs": ["timesync", "time"],
+       "slugs": ["timesync", "time"],Drum
```
Comment (athai) [inline source/draft_api.rst]: Drum?

File: source/draft_api.rst [hunk 3]
```diff
@@ -1021,20 +1028,47 @@ may have multiple members, spectators, and managers.
...
 
-POST and DELETE Endpoints
-~~~~~~~~~~~~~~~~~~~~~~~~~
+GET /times/:uuid follows the same rules (i.e. it will return the time if that time would
+be in the results of /times, or Authentication Failure otherwise).
+
+POST Endpoints
+~~~~~~~~~~~~~~
+
+POST /activites and POST /activities/:slug can be accessed by sitewide managers
```
Comment (athai) [inline source/draft_api.rst]: Misspelled activities

File: source/draft_api.rst [hunk 4]
```diff
@@ -1021,20 +1028,47 @@ may have multiple members, spectators, and managers.
...
+
+POST /project is accessible to sitewide managers
```
Comment (athai) [inline source/draft_api.rst]: Missed an 's' on the end of the projects endpoint

File: source/draft_api.rst [hunk 5]
```diff
@@ -1021,20 +1028,47 @@ may have multiple members, spectators, and managers.
...
+
+POST /project/:slug is accessible to the project's manager(s) and sitewide managers.
```
Comment (athai) [inline source/draft_api.rst]: Here as well.

**Comment by pop** [comment]
+1 Looks good to me.

**Comment by athai** [comment]
+1

**Pull request merged**